packages/domain/lib/src/usecases/base/in_out_usecase.dart

```diff
@@ -0,0 +1,9 @@
+abstract class InOutUseCase<In, Out, R> {
+  final R _repository;
+
+  InOutUseCase(this._repository);
```

Comment:
not the best solution to pass repository as generic. why? because not each usecase needs repository. so this type is redundant